lib/rmt/cli/repos_custom.rb

```diff
@@ -0,0 +1,124 @@
+class RMT::CLI::ReposCustom < RMT::CLI::Base
+
+  include ::RMT::CLI::ArrayPrintable
+
+  desc 'add URL NAME', 'Creates a custom repository.'
+  def add(url, name)
+    url += '/' unless url.end_with?('/')
+
+    previous_repository = Repository.find_by(external_url: url)
+
+    if previous_repository
+      raise RMT::CLI::Error.new('A repository by this URL already exists.')
+    end
+
+    repository_service.create_repository!(nil, url, {
+      name: name,
+      mirroring_enabled: 1,
+      autorefresh: 1,
+      enabled: 0
+    }, custom: true)
+
+    puts 'Successfully added custom repository.'
+  end
+
+  desc 'list', 'List all custom repositories'
+  def list
+    repositories = Repository.only_custom
+
+    raise RMT::CLI::Error.new('No custom repositories found.') if repositories.empty?
+
+    puts array_to_table(repositories, {
+      id: 'ID',
+      name: 'Name',
+      external_url: 'URL',
+      enabled: 'Mandatory?',
+      mirroring_enabled: 'Mirror?',
+      last_mirrored_at: 'Last Mirrored'
+    })
+  end
+  map ls: :list
+
+  desc 'enable ID', 'Enable mirroring of custom repository by ID'
+  def enable(id)
+    change_mirroring(id, true)
+  end
+
+  desc 'disable ID', 'Disable mirroring of custom repository by ID'
+  def disable(id)
+    change_mirroring(id, false)
+  end
+
+  desc 'remove ID', 'Remove a custom repository'
+  def remove(id)
+    repository = find_repository!(id)
+    repository.destroy!
+
+    puts "Removed custom repository by id \"#{id}\"."
+  end
+  map rm: :remove
+
+  desc 'products ID', 'Shows products attached to a custom repository'
+  def products(id)
+    repository = find_repository!(id)
+    products = repository.products
+
+    raise RMT::CLI::Error.new('No products attached to repository.') if products.empty?
+
+    puts array_to_table(products, {
+      id: 'Product ID',
+      name: 'Product Name'
+    })
+  end
+
+  desc 'attach ID PRODUCT_ID', 'Attach an existing custom repository to a product'
+  def attach(id, product_id)
+    product, repository = attach_or_detach(id, product_id)
```

Comment:
It isn't clear from the method name that attach_or_detach looks up a product and a repository from a pair of IDs.

Reply:
This wasn't my code, but you are totally right.

Reply:
This wasn't my code, but you are totally right.